lib/service/servicecfg/proxy.go

```diff
@@ -141,14 +141,18 @@ type ProxyConfig struct {
 // WebPublicAddr returns the address for the web endpoint on this proxy that
 // can be reached by clients.
 func (c ProxyConfig) WebPublicAddr() (string, error) {
-	return c.getDefaultAddr(c.WebAddr.Port(defaults.HTTPListenPort)), nil
+	return c.getDefaultAddr(c.WebAddr.Port(defaults.HTTPListenPort), true), nil
 }
 
-func (c ProxyConfig) getDefaultAddr(port int) string {
+func (c ProxyConfig) getDefaultAddr(port int, usePublicAddrPort bool) string {
```

Comment:
TBH I think that adding a flag here is confusing. I think that rather `WebPublicAddr()` function should determine the correct port (i.e. prefer public addr over web listen addr) and pass it here. Then you wouldn't also need to change kube addr method below.